### ISSUE TITLE
Remove drug mapping from the evidence source parsers

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/issues#3401]
As a **developer** I want **to remove the drug index dependency from the evidence source parsers** because **it creates a problem of circularity between the pre ETL and the ETL nodes**.

## Background

The cancer biomarkers module from the evidence source parsers required drug mapping when the data team introduced it. This required generating a look up table between drug names and IDs to fully annotate the biomarkers dataset. Here we introduced a problem where an output of the ETL was an input of a pre ETL step. The temporary solution was to use the drug index from the previous release to generate the pre ETL data (more context [here](https://github.com/opentargets/issues/issues/1821)).

Since then, the ETL has recently added a module to map drug names to their IDs since we had the exact same case when we introduced the `pharmacogenetics` dataset. Since this mapping logic is in place already, the ETL should implement it at the evidence level to be able to map any evidence where there is a drug name in `drugFromSource` to a `drugId` (as long as it is part of the index).

@DSuveges has recently realised that the biomarkers dataset was based on an outdated drug index (22.11). We are in the position now to phase out this dependency from the pre ETL steps. 

## Tasks

- [ ] BE/ETL: Use drug mapping step implemented [here](https://github.com/opentargets/platform-etl-backend/pull/329) to the evidence step
- [ ] Data: Remove drug index dependency from the cancer biomarkers module

## Acceptance tests

How do we know the task is complete? 

1. When I see the evidence widget between `breast cancer/BRCA2`, cisplatin is linked to `CHEMBL11359`

**Comment by prashantuniyal02** [comment]
Hey @ireneisdoomed , is this still relevant?

**Comment by ireneisdoomed** [comment]
Yes, will be resolved once we integrate the parsers into the orchestration

**Comment by d0choa** [comment]
Completed after the work in 26.03 regarding clinical data
